### PR TITLE
Adding changes for lifecycle.

### PR DESCRIPTION
**Pull request by peteclark-ft**
Master has been reset to the Production version, so that this bug fix can be pushed out independently.

Changes: 

Adding the lifecycle property to all relationships written by this component; it will be "annotations-v1" or "annotations-v2" for the v1/v2 components respectively.

### CODE REVIEW

File: annotations/cypher.go [hunk 1]
```diff
@@ -163,8 +158,11 @@ func (s service) Count() (int, error) {
 	}{}
 
 	query := &neoism.CypherQuery{
-		Statement:  `MATCH ()-[r{platformVersion:{platformVersion}}]->() RETURN count(r) as c`,
-		Parameters: neoism.Props{"platformVersion": s.platformVersion},
+		Statement: `MATCH ()-[r{platformVersion:{platformVersion}}]->()
+								 WHERE r.lifecycle = {lifecycle}
```
Comment (BerniVarga) [inline annotations/cypher.go]: Could you align these lines with the first one, please? pretty hard to read it this way :"> (sorry for this annoying type of comment)

File: annotations/cypher_test.go [hunk 2]
```diff
@@ -137,41 +139,88 @@ func TestWriteAllValuesPresent(t *testing.T) {
...
 		},
 	}
 
-	annotationsDriver.cypherRunner.CypherBatch([]*neoism.CypherQuery{createBrandQuery})
+	err := annotationsDriver.cypherRunner.CypherBatch([]*neoism.CypherQuery{contentQuery})
 
-	createContentQuery := &neoism.CypherQuery{
-		Statement: `MERGE (c:Content{uuid:{contentUuid}}) SET c :Thing RETURN c.uuid`,
+	annotationsToWrite := annotations{annotation{
+		Thing: thing{ID: getURI(conceptUUID),
```
Comment (BerniVarga) [inline annotations/cypher_test.go]: Could we have this in a file or in a separate struct? For readability - again. This test got quite big.

File: annotations/cypher_test.go [hunk 3]
```diff
@@ -137,41 +139,88 @@ func TestWriteAllValuesPresent(t *testing.T) {
 	cleanUp(t, contentUUID, []string{conceptUUID})
 }
 
-func TestWriteDoesNotRemoveExistingIsClassifedByBrandRelationships(t *testing.T) {
+func TestWriteDoesNotRemoveExistingIsClassifedByBrandRelationshipsWithoutLifecycle(t *testing.T) {
```
Comment (sjwells) [inline annotations/cypher_test.go]: 'Classified' rather than 'Classifed'

File: annotations/cypher.go [hunk 4]
```diff
@@ -309,7 +308,8 @@ func dropAllAnnotationsQuery(contentUUID string, platformVersion string) *neoism
 
 	var matchStmtTemplate string
 
-	//TODO hard-coded verification:
+	// TODO hard-coded verification:
+	// WE STILL NEED THIS UNTIL EVERYTHNG HAS A LIFECYCLE PROPERTY!
```
Comment (sjwells) [inline annotations/cypher.go]: 'EVERYTHING' not 'EVERYTHNG'

File: annotations/cypher_test.go [hunk 5]
```diff
@@ -137,41 +139,88 @@ func TestWriteAllValuesPresent(t *testing.T) {
...
 	assert := assert.New(t)
 
 	annotationsDriver = getAnnotationsService(t, v2PlatformVersion)
+	defer cleanDB(t, assert)
 
-	createBrandQuery := &neoism.CypherQuery{
-		Statement: `MERGE (b:Brand{uuid:{brandUuid}}) SET b :Concept:Thing RETURN b.uuid`,
+	contentQuery := &neoism.CypherQuery{
```
Comment (sjwells) [inline annotations/cypher_test.go]: is it a content query? or is it setting up the relationship between content and brand - 'createIsClassifiedByRelationship'?

File: annotations/cypher_test.go [hunk 6]
```diff
@@ -137,41 +139,88 @@ func TestWriteAllValuesPresent(t *testing.T) {
...
+					score{ScoringSystem: confidenceScoringSystem, Value: 0.8},
+				},
+				AgentRole: "http://api.ft.com/things/0edd3c31-1fd0-4ef6-9230-8d545be3880a",
+				AtTime:    "2016-01-01T19:43:47.314Z",
+			},
+		},
+	}}
+
+	assert.NoError(annotationsDriver.Write(contentUUID, annotationsToWrite), "Failed to write annotation")
+	checkRelationship(assert, contentUUID, "v2")
```
Comment (sjwells) [inline annotations/cypher_test.go]: checkIsClassifiedByRelationshipExistsForLifecycle

I am being picky, but it makes it easier to work out exactly what is being checked here.

File: annotations/cypher_test.go [hunk 7]
```diff
@@ -214,21 +265,6 @@ func TestWriteDoesNotRemoveExistingIsClassifedByBrandRelationships(t *testing.T)
...
-		},
-	}
-
-	annotationsDriver.cypherRunner.CypherBatch([]*neoism.CypherQuery{removeRelationshipQuery})
-
-	err = deleteNode(annotationsDriver, brandUUID)
-	assert.NoError(err, "Error trying to delete concept node with uuid %s, err=%v", brandUUID, err)
 }
 
 func TestWriteDoesRemoveExistingIsClassifedForV1TermsAndTheirRelationships(t *testing.T) {
```
Comment (sjwells) [inline annotations/cypher_test.go]: Another Classifed

File: annotations/cypher_test.go [hunk 8]
```diff
@@ -588,6 +624,25 @@ func readAnnotationsForContentUUIDAndCheckKeyFieldsMatch(t *testing.T, contentUU
 	}
 }
 
+func checkRelationship(assert *assert.Assertions, contentID string, platformVersion string) {
+	countQuery := `Match (t:Thing {uuid: {contentID}})-[r {lifecycle: {lifecycle}}]-(x) return count(r) as c`
```
Comment (sjwells) [inline annotations/cypher_test.go]: Would this break if another relationship with lifecycle, other than IS_CLASSIFIED, started to be used?

**Comment by sjwells** [comment]
Very minor comments but happy with the general thing :+1:

**Comment by BerniVarga** [comment]
+1

File: annotations/cypher_test.go (hunk 8 above)
Comment (peteclark-ft) [inline annotations/cypher_test.go]: I don't think so no - the relationship type should be irrelevant; we just expect any annotations written by this component to contain that property

**Pull request merged**